.evergreen/config.yml.in

```diff
@@ -730,6 +730,47 @@ functions:
       params:
         file: "src/xunit.xml"
 
+  "upload coverage report":
+    - command: shell.exec
+      params:
+        working_dir: "src"
+        script: |
+          ${PREPARE_SHELL}
+          npx nyc report --reporter=json
+    - command: s3.put
+      params:
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        local_file:  src/coverage/coverage-final.json
+        optional: true
+        # Upload the coverage report for all tasks in a single build to the same directory.
+        # TODO NODE-3897 - change upload directory to mongo-node-driver-next
+        # This change will require changing the `download and merge coverage` func as well
+        remote_file: mongo-node-driver/${revision}/${version_id}/coverage.${build_variant}.${task_name}.json
+        bucket: mciuploads
+        permissions: public-read
+        content_type: application/json
+        display_name: "Raw Coverage Report"
+
+  "download and merge coverage":
+    - command: shell.exec
+      params:
+        silent: true
+        working_dir: "src"
+        script: |
+          ${PREPARE_SHELL}
+          export AWS_ACCESS_KEY_ID=${aws_key}
+          export AWS_SECRET_ACCESS_KEY=${aws_secret}
+          # Download all the task coverage files.
+          # TODO NODE-3897 - finish this function.  the code below this point is untested because
+          #   aws s3 cp fails due to permissions errors
+          aws s3 cp --recursive s3://mciuploads/mongo-node-driver/${revision}/${version_id}/ coverage/
+
+          npx nyc merge coverage/ merged-coverage/coverage.json
+          npx nyc report -t merged-coverage --reporter=lcov --report-dir output
```

Comment:
Do we want lcov, or just html here?

Reply:
LCOV generates an html output too - https://istanbul.js.org/docs/advanced/alternative-reporters/#lcov

Reply:
I know it hurts precisely no one, should we just be explicit about html here if we're not going to use the lcov? but this is such a minor point no need to change if you think its useful to have it hang around

Reply:
Done!